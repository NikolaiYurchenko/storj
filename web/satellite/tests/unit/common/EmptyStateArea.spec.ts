// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { mount, shallowMount } from '@vue/test-utils';
import EmptyStateArea from '@/components/common/EmptyStateArea.vue';
import * as sinon from 'sinon';

describe('EmptyStateArea.vue', () => {

    it('renders correctly', () => {

        const wrapper = shallowMount(EmptyStateArea);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly without button props', () => {
        let mainTitle = 'testMainTitle';
        let additionalTextTag = '<h2>testAdditionalText</h2>';
        let additionalText = 'testAdditionalText';

        const wrapper = shallowMount(EmptyStateArea, {
            propsData: {mainTitle, additionalText: additionalTextTag},
        });

        let wrapperAdditionalTextTag = wrapper.find('h2');
        let wrapperMainTitleTag = wrapper.find('h1');

        expect(wrapperAdditionalTextTag.text()).toBe(additionalText);
        expect(wrapperMainTitleTag.text()).toBe(mainTitle);
    });

    it('renders correctly with isButtonShown prop', () => {
        let buttonLabel = 'testButtonLabel';

        const wrapper = shallowMount(EmptyStateArea, {
            propsData: {isButtonShown: true, buttonLabel, },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('trigger on press correctly with onButtonClick prop', () => {
        let onButtonClickSpy = sinon.spy();

        const wrapper = mount(EmptyStateArea, {
            propsData: {isButtonShown: true, onButtonClick: onButtonClickSpy},
        });

        wrapper.find('div.container').trigger('click');

        expect(onButtonClickSpy.callCount).toBe(1);
    });
});
