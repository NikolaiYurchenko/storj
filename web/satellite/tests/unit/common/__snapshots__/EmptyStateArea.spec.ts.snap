// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyStateArea.vue renders correctly 1`] = `
<div class="empty-state">
  <div class="empty-state__wrap">
    <h1></h1>
    <div class="empty-state__wrap__additional-text"></div>
    <!---->
    <div class="empty-state__wrap__img"></div>
  </div>
</div>
`;

exports[`EmptyStateArea.vue renders correctly with isButtonShown prop 1`] = `
<div class="empty-state">
  <div class="empty-state__wrap">
    <h1></h1>
    <div class="empty-state__wrap__additional-text"></div>
    <div id="addApiKeysPopupEmptyButton">
      <button-stub label="testButtonLabel" width="190px" height="50px" onpress="() => {
          /* istanbul ignore next */cov_24rw75ro5f.f[1]++;
          cov_24rw75ro5f.s[8]++;

          console.error('onPress is not reinitialized');
        }"></button-stub>
    </div>
    <div class="empty-state__wrap__img"></div>
  </div>
</div>
`;
